### PR TITLE
Cereal for TMRegion

### PR DESCRIPTION
**Pull request by dkeeney**
This covers Cereal serialization for TMRegion.

It is pending on #419

### CODE REVIEW

**Comment by dkeeney** [comment]
@breznak, @ctrl-z-9000-times, 
I am going to take a few days off, drive up to Oregon and visit some relatives.  It's great being retired and I can just take time off whenever I feel I need a break.

I pushed what I have for this PR.  After PR #422 is pushed to master this should build as well.  I will try to do this before I leave.

Lets see...other things on my plate:
- VectorFileEffector/VectorFileSensor  (coded but may still need some work)  #306
- remove the old serialization
- fix Python Region serialization (pickle) in the .py interface (PyBindRegion). #160
- Clean up some of the .py tests that were commented out due to serialization.
- Move MurmurHash3 to external #468
- Implement NetworkAPI example (in Python and C++) #460
- add a check for CMake in setup.py #436
- Change nupic namespace to htm  and engine_internal to network_api  #387
- create PyPI entry.  #19
- add a NetworkAPI  region for SDRClassifier. #309

**Comment by breznak** [comment]
>  drive up to Oregon and visit some relatives. 

enjoy the road-trip and have a great time! 

> create PyPI entry. #19 

it's almost half of the year, I think we should slowly be prioritizing the python 2 obsoletion, ie merging new py code and making sure it works, continued work on the bindings (and their testing), ...

**Comment by ctrl-z-9000-times** [comment]
Have a nice trip!

File: external/CMakeLists.txt [hunk 1]
```diff
@@ -107,3 +107,7 @@ include(mnist_data.cmake)
 # Cereal serialization package
 include(cereal.cmake)
 
+##################
+# MurmurHash3
+include(murmurhash.cmake)
```
Comment (breznak) [inline external/CMakeLists.txt]: nit, this could have been a simple separate PR. but good as is now

File: src/nupic/algorithms/Connections.cpp [hunk 2]
```diff
@@ -557,42 +557,6 @@ void Connections::bumpSegment(const Segment segment, const Permanence delta) {
 }
 
 
-void Connections::destroyMinPermanenceSynapses(
-                              const Segment segment, Int nDestroy,
```
Comment (breznak) [inline src/nupic/algorithms/Connections.cpp]: this is some mistake, was added by #446

Comment (breznak) [reply src/nupic/algorithms/Connections.cpp]: ..all the connections changes seem out of place. try `git checkout -- master src/nupic/..../Connections.* `

**Comment by dkeeney** [comment]
This branch is really messed up.  Don't know what happened.
I am going to start over with a clean branch....I have a few minutes before we leave.

**Pull request closed without merging**